### PR TITLE
[JBIDE-25905] route page: reuse existing table item instances

### PR DESCRIPTION
**Pull request by adietish**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
@jeffmaury @bdshadow this jira was caused by weird bugs in jface databinding: 
* the implementation was using ObservableMapLabelProvider to label data entries. This label provider works well as long as the input dataset stays the same. As soon as it's replaced (as it happens when one is pressing "Reset" in the UI), it wont update its cache (attributeMaps) and thus wont find the data that it wants to create labels for (empty labels in the screenshots attached to jira).  So I removed ObservableMapLabelProvider and replaced it with CellLabelProviders (which is what we use in other places in the tooling). The labeling then works fine but unveils another databdinging bug:
* Changes in boolean properties for an entry is not detected as a change and thus wont trigger a refresh in the tableviewer. I worked around this by listening to changes to the routing port property, manually triggering a refresh of the tableviewer.

**Comment by adietish** [comment]
@jeffmaury @bdshadow please review

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/PodPortValidator.java [hunk 1]
```diff
@@ -16,73 +16,93 @@
...
+			status = validateValue(port);
+			if (status.isOK()) {
+				status = validateUniqueness(port);
+			}
+		}
+		return status;
+	}
+
+	private IStatus validateUniqueness(final String port) {
+		for (IServicePort existing : ports) {
```
Comment (bdshadow) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/PodPortValidator.java]: can be simpler:`if ports.contains(existing.getTargetPort())`

Comment (adietish) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/PodPortValidator.java]: @bdshadow this unfortunately wont work. ports contains IServicePort objects. What we get as parameter is a string. We need to check if among the service ports objects, there's already one whose target port matches the given (String) parameter.

**Pull request merged**